src/main/java/Toothless.java

```diff
@@ -0,0 +1,339 @@
+        }
+    }
+
+    /**
+     * Returns the data of the task to save in the text file.
+     * @param task The task to be saved.
+     * @return The data of the task to be saved in the text file.
+     */
+    private String getTaskData(Task task) {
+        String taskType = task instanceof ToDos ? "T" :
+                task instanceof Deadline ? "D" :
+                        task instanceof Events ? "E" : "";
+        String baseInfo = taskType + " | " + (task.isDone ? "1" : "0") + " | " + task.description;
+
+        if (task instanceof Deadline) {
+            return baseInfo + " | " + ((Deadline) task).deadline;
+        } else if (task instanceof Events) {
+            Events event = (Events) task;
+            return baseInfo + " | " + event.eventStart + " | " + event.eventEnd;
+        }
+        return baseInfo;
+    }
+
+    /**
+     * Loads the tasks from a file.
+     */
+    public void loadTasks() {
+        list.clear();
+        try (BufferedReader reader = new BufferedReader(new FileReader(DATA_FILE_PATH))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                Task task = parseDataToTask(line);
+                if (task != null) {
+                    list.add(task);
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Toothless:\nOh no! There is an error loading quests from file.");
+        }
+    }
+
+    /**
+     * Parses the data from the file to a task.
+     * @param taskData The data of the task.
+     * @return The task parsed from the data.
+     */
+    private Task parseDataToTask(String taskData) {
+        String[] splitData = taskData.split(" \\| ");
+
+        String taskType = splitData[0];
+        boolean isDone = splitData[1].equals("1");
+        String description = splitData[2];
+
+        switch(taskType) {
+            case "T":
+                return new ToDos(description, isDone);
+            case "D":
+                String deadline = splitData[3];
+                return new Deadline(description, LocalDateTime.parse(deadline), isDone);
+            case "E":
+                String eventStart = splitData[3];
+                String eventEnd = splitData[4];
+                return new Events(description, LocalDateTime.parse(eventStart), LocalDateTime.parse(eventEnd), isDone);
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Begins the chat application.
+     * The user can add tasks to the task list.
+     */
+    public void beginChat() throws ToothlessExceptions {
+        loadTasks();
+
+        System.out.println("Welcome to the dragon's den!\n" + TOOTHLESS_LOGO);
+        System.out.println("Toothless:\n" +
+                "Greetings, Dragon Rider!\n\n" +
+                "I'm Toothless, your friendly dragon companion.\n" +
+                "What adventure shall we embark on today?\n\n" + DIVIDER);
```

Comment:
I now know where ToothlessException comes from! But for readability, it would be much better to rename the exception.